src/loading.ts

```diff
@@ -0,0 +1,73 @@
+import { OperatorFunction, merge, combineLatest, of } from 'rxjs'
+import { share, startWith, map, filter, mapTo, delay, endWith, scan } from 'rxjs/operators'
+import { isEqual } from 'lodash'
+
+export const LOADING = 'loading' as const
+
+/**
+ * An emission from a result provider.
+ *
+ * @template T The type of the result. Should typically include an empty value, or even an error type.
+ */
+export interface MaybeLoadingResult<T> {
+    /**
+     * Whether the result provider is currently getting a new result.
+     */
+    isLoading: boolean
+
+    /**
+     * The latest result.
+     */
+    result: T
+}
+
+/**
+ * Handles input of MaybeLoadingResult (which contains both results and loading states) and returns a sequence of clear
+ * instructions on when to show a loader, results or nothing.
+ *
+ * @param loaderDelay After how many milliseconds of no results a loader should be shown.
+ * @param emptyResultValue The value that represents no results. This will be emitted, and also deep-compared to with `isEqual()`. Example: `null`, `[]`
```

Comment:
```suggestion
 * @param emptyResultValue The value that represents the absence of results. This will be emitted, and also deep-compared to with `isEqual()`. Example: `null`, `[]`
```